docs/audit-log-filter-variables.md

```diff
@@ -0,0 +1,772 @@
+
+!!! important
+
+    Avoid using this function. This function performs an operation that is similar to uninstalling and reinstalling the plugin. Filters are detached from all current sessions. To restart logging, the current sessions must either disconnect and reconnect or do a change-user operation. 
+
+#### Parameters
+
+None.
+
+#### Returns
+
+This function returns either an `OK` for success or an error message for failure.
+
+#### Example
+
+```{.bash data-prompt="mysql>"}
+mysql> SELECT audit_log_filter_flush();
+```
+
+??? example "Expected output"
+
+    ```{.text .no-copy}
+    +--------------------------+
+    | audit_log_filter_flush() |
+    +--------------------------+
+    | OK                       |
+    +--------------------------+
+    ```
+
+### audit_log_read()
+
+If the audit log filter format is JSON, this function reads the audit log and returns an array of the audit events as a JSON string. Generates an error if the format is not JSON.
+
+#### Parameters
+
+None. If the start position is not provided, the read continues from the current position. 
+
+Optional: You can specify a starting position for the read with `start` or a `timestamp` and an `id`, both items are considered a bookmark and can be used to identify an event. You must include both (`timestamp` and `id`) or an error is generated. If the `timestamp` does not include a `time` section, the function assumes  the time is `00:00`.
+
+You can also provide a `max_array_length` to limit the number of log events.
+
+Call[`audit_log_read_bookmark()`](#audit_log_read_bookmark) to return the most recently written event.
+
+#### Returns
+
+This function returns a string of a JSON array of the audit events, or a JSON NULL value. Returns `NULL` and generates an error if the call fails.
+
+#### Example
+
+```{.bash data-prompt="mysql>"}
+mysql> SELECT audit_log_read(auditK_log_read_bookmark());
+```
+
+??? example "Expected output"
+
+    ```{.text .no-copy}
+    +------------------------------------------------------------------------------+
+    | audit_log_read(audit_log_read_bookmark())                                   |
+    +------------------------------------------------------------------------------+
+    | [{"timestamp" : "2023-06-02 09:43:25", "id": 10,"class":"connection",]       |
+    +------------------------------------------------------------------------------+
+    ```
+
+### audit_log_read_bookmark()
+
+This function provides a bookmark for the most recently written audit log event as a JSON string. Generates an error if the format is not JSON.
+
+If this function is used with [`audit_log_read()](#audit_log_read), the `audit_log_read()` function starts reading at that position.
+
+#### Parameters
+
+None.
+
+#### Returns
+
+This function returns a string of a JSON array of the audit events, or a JSON NULL value. Returns `NULL` and generates an error if the call fails.
```

Comment:
A JSON string containing a bookmark for success, or NULL and an error for failure.